beacon-node-oapi.yaml

```diff
@@ -6,10 +6,24 @@ info:
     API specification for the beacon node, which enables users to query and participate in Ethereum 2.0 phase 0 beacon chain.
 
     All requests by default send and receive JSON, and as such should have either or both of the "Content-Type: application/json"
-    and "Accept: application/json" headers.
-    In addition, some request can return data in the SSZ format.
-    To indicate that SSZ data is required in response to a request the header "Accept: application/octet-stream" should be sent.
-    Note that only a subset of requests can respond with data in SSZ format; these are noted in each individual request.
+    and "Accept: application/json" headers.  In addition, some requests can return data in the SSZ format.  To indicate that SSZ
+    data is required in response to a request the header "Accept: application/octet-stream" should be sent.  Note that only a subset
+    of requests can respond with data in SSZ format; these are noted in each individual request.
+
+    API endpoints are individually versioned.  As such, there is no direct relationship between all v1 endpoints, all v2 endpoints,
+    _etc._ and no such relationship should be inferred.  The rules that require an increase in version number are as follows:
+
+      - no field that is listed in an endpoint shall be removed without an increase in the version number
+      - no field that is listed in an endpoint shall be altered terms of format (_e.g._ from a string to an array) without an
+        increase in the version number
+
+    Note that it is possible for a field to be added to an endpoint without an increase in the version number.
+
+    All JSON responses return the requested data under a "data" key in the top level of their response.  Additional metadata may or
+    may not be present in other keys at the top level of the response.  Metadata's presence or absence can be dependent on the type
+    and version of the beacon node, the state of the chain, and other conditions.  Metadata is not versioned, and as such the same
+    version of an API endpoint may return different metadata at different times.
```

Comment:
If metadata is used for anything and we allow removal or renaming metadata fields without a version bump then it would make for an unreliable API schema and probably introduce breaking changes. 

Personally, I would like to see metadata be subject to the same versioning rules as data fields.

Reply:
I'd say we should be able to add metadata fields, but not remove.  Having to bump the version on every single change just breaks all clients unnecessarily.

Personally I'd apply the same policy to data - ignore unknown fields is the bedrock of internet compatibility - but it doesn't seem like I'm going to get my way on that...

Reply:
I'd prefer 'safe to add' type mantra, as otherwise we're going to potentially have a lot of versioning in our future...

If a client doesn't find a field they need, then they should handle it or fail depending on case, and if we do have to add something that makes things not compatible, then we could version at that point in time...